content/chat/participant-activity.textile

```diff
@@ -1,7 +1,220 @@
+
+```
+{
+  "status-general": ["presence"],
+  "general": ["publish", "history", "subscribe"],
+}
+```
+
+Meanwhile the server can also subscribe to the chat participants channel with an api key having the following capabilities:
+
+```
+{
+  "status-general": ["presence", "subscribe"],
+  "general": ["publish", "history", "subscribe"],
+}
+```
+
+h2(#last-seen). Last seen
+
+Along with determining whether a chat participant is actively online or not, another key piece of functionality in a chat application is to determine when the chat participant was last seen as online. So long as the client, the chat participant is using specifically, calls @enter@ on the presence set for that channel, you'll have a historical record of that entry and a record of when the chat participant left the presence set. Whether it was specifically calling @leave@ or they timed out due to an unstable connection.
+
+To retrieve the channel's presence history, call the channel's presence followed by its history function. Calling @history@ will retrieve up to 1000 records, starting from the latest presence entries. To find the @leave@ events for a specific chat participant, loop through each entry, first filtering out any records by the @clientId@ of your desired chat participant, and the action, which is @leave@. You then need to sort the records, finding the latest timestamp first. Below is an example of how you would implement this functionality:
+
+```[javascript]
+channel.attach(() => {
+  const presence = channel.presence;
+
+  presence.history({}, (err, resultPage) => {
+      const participantStatus = [];
+
+      if (err) {
+        console.log('Unable to get presence history; err = ' + err.message);
+      } else {
+        participantStatus = resultPage.items.filter(
+          item => item.clientId === 'PARTICIPANT_CLIENT_ID' && item.action === 'leave'
+        );
+
+        participantStatus.sort((recordA, recordB) => {
+          return recordA.timestamp > recordB.timestamp;
+        });
+      }
+  });
+});
+```
+
+h2(#remain-present-for). Keep connected for period of time before removing chat participant
+
+Clients that are part of a chat participant presence set remain present, by default, for 15 seconds after they are abruptly disconnected. For example, when the internet connection suddenly drops or the client is changing networks. This delay is to avoid repeated @leave@ and @enter@ events being sent when a client is experiencing an unstable connection. A different disconnection timeout can be defined in the @remainPresentFor@ client configuration.
```

Comment:
Switched back to 'chat participant presence set', so I think we really need to nail the consistent terminology here.